src/main/java/jenkins/plugins/logstash/persistence/LogzioDao.java

```diff
@@ -0,0 +1,127 @@
+package jenkins.plugins.logstash.persistence;
+
+import com.github.wnameless.json.flattener.JsonFlattener;
+import com.google.common.io.Files;
+
+import io.logz.sender.LogzioSender;
+import io.logz.sender.SenderStatusReporter;
+import io.logz.sender.com.google.gson.JsonObject;
+import io.logz.sender.exceptions.LogzioParameterErrorException;
+
+import java.io.*;
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import jenkins.model.Jenkins;
+import jenkins.plugins.logstash.LogstashConfiguration;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+/**
+ * Logz.io Data Access Object.
+ *
+ * @author Ido Halevi
+ */
+
+public class LogzioDao extends AbstractLogstashIndexerDao {
+    private static final int CONNECT_TIMEOUT = 10*1000;
+    private static final int CORE_POOL_SIZE = 2;
+    private static final int DRAIN_TIMEOUT = 2;
+    private static final int FS_PERCENT_THRESHOLD = 99;
+    private static final int GC_PERSISTED_QUEUE_FILE_INTERVAL_SECOND = 30;
+    private static final int SOCKET_TIMEOUT = 10*1000;
+    private static final String TYPE = "jenkins_plugin";
+    private final LogzioSender logzioSender;
+
+    private String key;
+    private String host;
+
+    //primary constructor used by indexer factory
+    public LogzioDao(String host, String key){
+        this(null, host, key);
+    }
+
+    // Factored for unit testing
+    LogzioDao(LogzioSender factory, String host, String key){
+        this.host = host;
+        this.key = key;
+
+        // create file for sender queue
+        File fp;
+        Jenkins jenkins = Jenkins.getInstanceOrNull();
+        if (jenkins == null){
```

Comment:
You encountered a situation where Jenkins was null (maybe during unit tests?)
Builds should only start when Jenkins is fully up.